File: Iniciante/1049.rs
use std::io;

fn main() {
  let mut input1 = String::new();
  let mut input2 = String::new();
  let mut input3 = String::new();

  io::stdin().read_line(&mut input1).unwrap();
  io::stdin().read_line(&mut input2).unwrap();
  io::stdin().read_line(&mut input3).unwrap();

  input1 = input1.trim().to_string();
  input2 = input2.trim().to_string();
  input3 = input3.trim().to_string();

  let mut result = String::new();
  if input1 == "vertebrado" {
    if input2 == "ave" {
      if input3 == "carnivoro" {
        result.push_str("aguia");
      } else {
        result.push_str("pomba");
      }
    } else {
      if input3 == "onivoro" {
        result.push_str("homem");
      } else {
        result.push_str("vaca");
      }
    }
  } else {
    if input2 == "inseto" {
      if input3 == "hematofago" {
        result.push_str("pulga");
      } else {
        result.push_str("lagarta");
      }
    } else {
      if input3 == "hematofago" {
        result.push_str("sanguessuga");
      } else {
        result.push_str("minhoca");
      }
    }
  }
  println!("{}",result);
}

File: Iniciante/1049/1049.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
  public static void main(String[] args) throws IOException {
    InputStreamReader stream = new InputStreamReader(System.in);
    BufferedReader reader = new BufferedReader(stream);

    String line1 = reader.readLine();
    String line2 = reader.readLine();
    String line3 = reader.readLine();

    String result;
    if (line1.equals("vertebrado")) {
      if (line2.equals("ave")) {

        if (line3.equals("carnivoro")) {
          result = "aguia";
        } else {
          result = "pomba";
        }

      } else if (line3.equals("onivoro")) {
        result = "homem";
      } else {
        result = "vaca";
      }

    } else if (line2.equals("inseto")) {
      if (line3.equals("hematofago")) {
        result = "pulga";
      } else {
        result = "lagarta";
      }
    } else if (line3.equals("hematofago")) {
      result = "sanguessuga";
    } else {
      result = "minhoca";
    }

    System.out.println(result);
  }
}
